app/src/main/java/org/simple/clinic/main/TheActivity.kt

```diff
@@ -307,21 +291,13 @@ class TheActivity : AppCompatActivity(), TheActivityUi {
   }
 
   override fun onBackPressed() {
-    val interceptCallback = screenRouter.offerBackPressToInterceptors()
-    if (interceptCallback.intercepted) {
-      return
-    }
-    val popCallback = screenRouter.pop()
-    if (popCallback.popped) {
-      return
+    if (!router.onBackPressed()) {
+      super.onBackPressed()
     }
-    super.onBackPressed()
   }
 
   override fun onSaveInstanceState(outState: Bundle) {
-    if (features.isEnabled(LogSavedStateSizes)) {
```

Comment:
We don't need to do that anymore. The older version were having issues because the saved instance state would be collected for all screens in the backstack at once whereas here, each screen is its own fragment.